swan/Dockerfile

```diff
@@ -0,0 +1,192 @@
+# Installs the extensions and packages required by SWAN
+# to adapt the base alma9 notebook image to SWAN.
+
+FROM gitlab-registry.cern.ch/pesteves/imageregistry:notebook
+
+LABEL maintainer="swan-admins@cern.ch"
+
+ARG NB_UID="1000"
+ARG BUILD_TAG=daily
+ENV VERSION_DOCKER_IMAGE=$BUILD_TAG
+
+RUN echo "Building systemuser image with tag ${VERSION_DOCKER_IMAGE}"
+
+# Switch to superuser to install packages
+USER root
+
+# Install required dnf packages
+RUN dnf install -y gcc \
+    gcc-c++ \
+    git \
+    kernel-devel \
+   # libcurl-openssl-devel \  -- Does not exist in dnf repos
+    libffi-devel \
+    ncurses-devel \
+    nano \
+    nodejs \
+    patch \
+    sqlite-devel \
+    unzip \
+    which \
+   # zeromq3-devel \ -- Does not exist in dnf repos
+    perl-Digest-MD5 \
+    texlive \
+    fontconfig && \
+    # Clear the dnf cache as we no longer need to install packages
+    dnf clean all && \
+    rm -rf /var/cache/dnf
+
+# Install a newer version of TeX Live, than the one available in dnf repos
+# For converting to PDF
+ENV PATH /usr/local/texlive/2023/bin/x86_64-linux/:$PATH
+RUN tlmgr install adjustbox \
+    tcolorbox \
+    environ \
+    trimspaces \
+    adjustbox \
+    collectbox \
+    ucs \
+    titling \
+    enumitem \
+    type1cm \
+    cm-super \
+    collection-fontsrecommended
+
+# Install python pip packages
+# RUN pip3 --no-cache-dir install \
+#           'ipyparallel==6.3.0' \
+#            'notebook==6.4.2' \
+#            'jupyterhub==4.0.2' \
+#            'urllib3==1.26.16' \
+#            'jupyterlab==3.0.17' \
+#            'jupyter_nbextensions_configurator' \
+#            'voila'
+
+# Install python pip packages
+RUN mamba install --yes \
+    'jupyter_nbextensions_configurator' \
+    'voila'
+
+VOLUME /notebooks
+WORKDIR /notebooks
+
+CMD ["jupyter", "notebook"]
+
+# Disable requiretty and secure path - required by systemuser.sh
+RUN sed -i'' '/Defaults \+requiretty/d'  /etc/sudoers && \
+    sed -i'' '/Defaults \+secure_path/d' /etc/sudoers
+
+# Install required packages
+RUN dnf install -y \
+    # Required by ROOT
+    libXft \
+    # Required by matplotlib
+    tk \
+    # Useful to monitor resource consumption
+    htop \
+    # Install Cloudera dependencies - required by IT Spark clusters
+    alsa-lib \
+    at \
+    bc \
+    cronie \
+    cvs \
+    # db4-cxx \ -- Is not present in DNF repos.
+    # db4-devel \ -- Is not present in DNF repos.
+    file \
+    gdbm-devel \
+    gettext \
+    jpackage-utils \
+    libXtst \
+    man \
+    passwd \
+    # pax \ -- Is not present in DNF repos.
+    perl-CGI \
+    perl-ExtUtils-MakeMaker \
+    perl-Test-Simple \
+    perl-devel \
+    rsyslog \
+    time \
+    xz-lzma-compat \
+    # Required by Geant4 (libXm)
+    openmotif \
+    # Required by Oracle
+    libaio \
+    # Required for key4hep
+    environment-modules 
+
+# Required by jupyter-resource-usage
+RUN pip install --no-deps --no-cache-dir psutil==5.8.0
```

Comment:
Not seeing the installation of `bs4`, `jupyter-server-proxy` (now can come direcly from a repo instead of git), `simpervisor`